packages/components/src/components/post-header/post-header.tsx

```diff
@@ -162,31 +148,13 @@ export class PostHeader {
     }
   }
 
-  private getScrollParent(): Element | Document {
-    let parent: Element | Document = this.host.parentElement;
-    if (parent.tagName === 'BODY') {
-      parent = document;
-    }
-    return parent;
-  }
-
   private handleScrollEvent() {
-    // Credits: "https://github.com/qeremy/so/blob/master/so.dom.js#L426"
-    const st = Math.max(
-      0,
-      this.scrollListenerElement instanceof Document
-        ? this.scrollListenerElement.documentElement.scrollTop
-        : this.scrollListenerElement.scrollTop,
-    );
-
-    this.host.style.setProperty('--header-scroll-top', `${st}px`);
+    this.host.style.setProperty('--header-scroll-top', `${this.scrollParent.scrollTop}px`);
```

Comment:
Using `this.scrollParent` getter here would mean it is executed anytime you scroll 1px. Getting the scroll parent in the `connectedCallback`, or at least throttling it might be more efficient.

Reply:
Can we do that in a separate ticket?

Reply:
I opened this ticket: #4794